### ISSUE TITLE
Clear cache and reload should probably explain what it actually does

### ISSUE THREAD

**Issue by pafcu** [vector-im/riot-web#5800]
### Description

Currently there is no explanation what the "Clear cache and reload" button in user settings _actually_ does. What is stored in the cache? What happens if you clear it? Currently the button is red, which indicates it might be dangerous, but the user has no way of deciding if they want to "take the risk". As per https://github.com/vector-im/riot-web/issues/5724#issuecomment-348536967 it could be made yellow/orange instead, but _some_ explanation of what the button does, and why you should be careful with it, would be nice to have.

### Version information

- **Platform**: web

For the web app:

- **Browser**: Firefox 57
- **OS**: Linux
- **URL**: riot.im/develop

**Comment by lampholder** [comment]
Ah yes, this adds a nuance to https://github.com/vector-im/riot-web/issues/7539

**Comment by anoadragon453** [comment]
![image](https://user-images.githubusercontent.com/1342360/68198917-e3a5c980-ffb4-11e9-8d0e-64ea8712cc26.png)

It really should get its own section. I've also had reports of users hitting "Clear cache and reload" accidentally when they go to hit "Submit debug logs", which is probably not what we want if we want users to catch hard-to-find bugs.

**Comment by abhinav-TB** [comment]
@aaronraimist I would like to work on this issue could you give some guidence

**Comment by aaronraimist** [comment]
@abhinav-TB This is where the button is created in code.  https://github.com/matrix-org/matrix-react-sdk/blob/4f52afdb3c15b5c46cf8a00f0b971637e1147148/src/components/views/settings/tabs/user/HelpUserSettingsTab.js#L273-L277

You can add a sentence or two explaining what clear cache does right above it. It would look something like https://github.com/matrix-org/matrix-react-sdk/blob/4f52afdb3c15b5c46cf8a00f0b971637e1147148/src/components/views/settings/tabs/user/HelpUserSettingsTab.js#L171-L174

If you need more help feel free to join the `#element-dev:matrix.org` room.

**Comment by salmankhan432423** [comment]
I need to create a web like the one mentioned in my link, which should be done by Laravel or Digango framework. This website will be just like https://alltricksusa.blogspot.com/2021/03/settimeoutfunction-window.html If you can do it for me then email here rhr@neoitbd.com

**Comment by salmankhan432423** [comment]
I need to create a web like the one mentioned in my link, which should be done by Laravel or Digango framework. This website will be just like https://alltricksusa.blogspot.com/2021/03/settimeoutfunction-window.html If you can do it for me then email here rhr@neoitbd.com

**Comment by tawfiek** [comment]
I would like to help with this, can anybody Assign it to me ?!

**Comment by SimonBrandner** [comment]
@tawfiek, done!

**Comment by tawfiek** [comment]
I think it's fixed already !!
![Screenshot from 2021-03-24 12-12-23](https://user-images.githubusercontent.com/27981868/112292909-39f2d580-8c9a-11eb-9f0e-0224b62f67da.png)

**Comment by SimonBrandner** [comment]
> I think it's fixed already !!

Is it? I don't think I understand...

**Comment by tawfiek** [comment]
compared to the 

> ![image](https://user-images.githubusercontent.com/1342360/68198917-e3a5c980-ffb4-11e9-8d0e-64ea8712cc26.png)
> 
> It really should get its own section. I've also had reports of users hitting "Clear cache and reload" accidentally when they go to hit "Submit debug logs", which is probably not what we want if we want users to catch hard-to-find bugs.
compared to this view I think this issue has been solved somewhere :octocat:

**Comment by SimonBrandner** [comment]
> "Clear cache and reload" button in user settings actually does. What is stored in the cache? What happens if you clear it? 

Yeah, but I don't think there is anything about this...

**Comment by ptman** [comment]
I don't know if this is the right place. Please direct me to a more suitable issue.

Is it possible to detect when the local state ("cache") is in conflict with what the server thinks is true and do a local state rebuild ("clear cache and reload") automatically?

**Comment by t3chguy** [comment]
> Is it possible to detect when the local state ("cache") is in conflict with what the server thinks is true and do a local state rebuild ("clear cache and reload") automatically?

Not without basically having to constantly initial sync with the server and comparing, this operation is really expensive on the server and processing that much JSON will likely also slow down your user experience.

**Comment by ptman** [comment]
@t3chguy Ok, let me rephrase: I'm not looking for a way to make sure the cache is always known to be good. But if some operation reveals that the cache is indeed out-of-sync. It would help even if it wasn't perfect.

**Comment by t3chguy** [comment]
You can of course write heuristics, but getting a good heuristic for this which doesn't have too many false positives will probably be an artform in itself. You're guessing at best, unless you ask the server for what it thinks and comparing, which will increase server load a *lot*.

**Comment by hifi** [comment]
There are two clear cases for cache inconsistency that are user facing:

 - joining a room you're already on
 - leaving a room you've already left

These can't be resolved with any other mean than clearing the cache AFAIK so when the client and server disagree on this the user could be presented with a choice: "Seems like your client state is not in sync with the server, we can reset your client automatically but it will take a while, proceed?" (rephrased better).

**Comment by Penguin-Guru** [comment]
It would be nice if this could also warn users how long the operation could take. I'm running one right now that's over an hour in. The window has been pure white with the same spinner in the center the entire time, so I assume it's working.

**Comment by t3chguy** [comment]
> It would be nice if this could also warn users how long the operation could take. 

Well the warning would have to be "If your server has insufficient resources it might take longer than you remain alive" given there's no way to know how long it will take

**Comment by Penguin-Guru** [comment]
> > It would be nice if this could also warn users how long the operation could take.
> 
> Well the warning would have to be "If your server has insufficient resources it might take longer than you remain alive" given there's no way to know how long it will take

That sounds good to me. In my case, I had to kill the process after letting it run for about 26 hours. It seems there was a malfunction but I can't be sure.
